### PR TITLE
VIH-6674 judge single & multi day emails

### PR DESCRIPTION
**Pull request by virensareen**
### JIRA link (if applicable) ###
VIH-6674
VIH-7158

### Change description ###
Enabling judge automation emails for:
- Single day confirmation
- Single day amendment

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by RuwanikaWeerakkody — APPROVED** [review]

**Review by shaed-parkar — CHANGES_REQUESTED** [review]

File: AdminWebsite/AdminWebsite.UnitTests/Services/HearingServiceTests.cs [hunk 1]
```diff
@@ -43,23 +46,49 @@ public async Task should_send_confirmation_email_to_all_participants_except_a_ju
...
+            _hearing.Other_information = JsonConvert.SerializeObject(new OtherInformationDetails {JudgeEmail = null});
+
+            var secondHearing = InitHearing();
+            secondHearing.Scheduled_date_time = _hearing.Scheduled_date_time.AddDays(1).AddHours(3).AddMinutes(20);
+            await _service.SendHearingUpdateEmail(_hearing, secondHearing);
+
+            _mocker.Mock<INotificationApiClient>()
+                .Verify(
+                    x => x.CreateNewNotificationAsync(It.IsAny<AddNotificationRequest>()),
+                    Times.Exactly(3));
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite.UnitTests/Services/HearingServiceTests.cs:69]: This test could break if someone changes the participant count. Maybe use the participant count - the judge

File: AdminWebsite/AdminWebsite.UnitTests/Services/HearingServiceTests.cs [hunk 2]
```diff
@@ -43,23 +46,49 @@ public async Task should_send_confirmation_email_to_all_participants_except_a_ju
...
-            var judge = _hearing.Participants.First(x => x.User_role_name == "Judge");
             var secondHearing = InitHearing();
+            secondHearing.Other_information = JsonConvert.SerializeObject(new OtherInformationDetails {JudgeEmail = "judge@hmcts.net"});
             secondHearing.Scheduled_date_time = _hearing.Scheduled_date_time.AddDays(1).AddHours(3).AddMinutes(20);
             await _service.SendHearingUpdateEmail(_hearing, secondHearing);
 
+            _mocker.Mock<INotificationApiClient>()
+                .Verify(
+                    x => x.CreateNewNotificationAsync(It.IsAny<AddNotificationRequest>()),
+                    Times.Exactly(4));
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite.UnitTests/Services/HearingServiceTests.cs:83]: Same here, participant count would be more reliable

File: AdminWebsite/AdminWebsite.UnitTests/Services/HearingServiceTests.cs [hunk 3]
```diff
@@ -43,23 +46,49 @@ public async Task should_send_confirmation_email_to_all_participants_except_a_ju
...
+        }
+        
+        [Test]
+        public async Task should_send_multiday_confirmation_email_to_all_participants()
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite.UnitTests/Services/HearingServiceTests.cs:87]: update the title so it's clear all but judges get the MDH confirmation email

File: AdminWebsite/AdminWebsite.UnitTests/Services/HearingServiceTests.cs [hunk 4]
```diff
@@ -43,23 +46,49 @@ public async Task should_send_confirmation_email_to_all_participants_except_a_ju
...
+        {
+            var judge = _hearing.Participants.First(x => x.User_role_name == "Judge");
+            await _service.SendMultiDayHearingConfirmationEmail(_hearing, 2);
+
             _mocker.Mock<INotificationApiClient>()
                 .Verify(
                     x => x.CreateNewNotificationAsync(It.Is<AddNotificationRequest>(r => r.ParticipantId != judge.Id)),
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite.UnitTests/Services/HearingServiceTests.cs:94]: again participant count might be more reliable minus the judge

**Review by shaed-parkar — COMMENTED** [review]
just a few more things to clean up

File: AdminWebsite/AdminWebsite.UnitTests/Services/HearingServiceTests.cs [hunk 5]
```diff
@@ -93,6 +121,32 @@ public async Task should_send_reminder_email_to_all_participants_except_a_judge(
...
+        [Test]
+        public async Task should_send_confirmation_email_to_judge_when_reminder_is_sent_to_participants()
+        {
+            var judge = _hearing.Participants.First(x => x.User_role_name == "Judge");
+            _hearing.Other_information = JsonConvert.SerializeObject(new OtherInformationDetails {JudgeEmail = "judge@hmcts.net"});
+            await _service.SendHearingReminderEmail(_hearing);
+
+            _mocker.Mock<INotificationApiClient>()
+                .Verify(
+                    x => x.CreateNewNotificationAsync(It.Is<AddNotificationRequest>(r => r.ParticipantId == judge.Id)),
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite.UnitTests/Services/HearingServiceTests.cs:134]: verify email type is judge confirmation

**Review by shaed-parkar — COMMENTED** [review]
Update assertion but pretty much done

**Pull request merged**